Deploy/java/s3-client-test-workspace/s3-client-test/src/main/java/gov/cms/ab2d/s3/S3ClientTest.java

```diff
@@ -0,0 +1,136 @@
+package gov.cms.ab2d.s3;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import software.amazon.awssdk.auth.credentials.EnvironmentVariableCredentialsProvider;
+import software.amazon.awssdk.auth.credentials.InstanceProfileCredentialsProvider;
+import software.amazon.awssdk.auth.credentials.ProfileCredentialsProvider;
+import software.amazon.awssdk.auth.credentials.SystemPropertyCredentialsProvider;
+import software.amazon.awssdk.core.ResponseInputStream;
+import software.amazon.awssdk.core.exception.SdkClientException;
+import software.amazon.awssdk.core.exception.SdkServiceException;
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.s3.model.GetObjectRequest;
+import software.amazon.awssdk.services.s3.model.GetObjectResponse;
+import software.amazon.awssdk.services.s3.model.ListBucketsResponse;
+import software.amazon.awssdk.services.s3.S3Client;
+
+import java.io.IOException;
+import java.io.InputStreamReader;
+
+class S3ClientTest
+{
+	private static final Logger log = LoggerFactory.getLogger(S3ClientTest.class);
+	
+	public static void main(String args[]) 
+	{
+		// Set region
+		String s3Region = "us-east-1";
+		final Region region = Region.of(s3Region);
+		
+		// Set S3 bucket
+		String s3Bucket = "ab2d-optout-data-dev";
+		
+		// Set S3 file name
+		String s3Filename = "T#EFT.ON.ACO.NGD1800.DPRF.D191029.T1135430";
+		
+		// Get s3 client for EC2 instance (if applicable)
+		S3Client s3Client = getS3ClientForEc2Instance(region, s3Bucket);
+		if (s3Client == null) {
+			// Get s3 client for development
+			s3Client = getS3ClientForDevelopment(region, s3Bucket);
+		}
+		
+		// build GetObjectRequest
+		final GetObjectRequest getObjectRequest = GetObjectRequest.builder()
+				.bucket(s3Bucket)
+				.key(s3Filename)
+				.build();
+		
+		try {
+			final ResponseInputStream<GetObjectResponse> responseInputStream = s3Client.getObject(getObjectRequest);
+			InputStreamReader inputStreamReader = new InputStreamReader(responseInputStream);
+			try {
+				inputStreamReader.close();
```

Comment:
This try-block to close the inputStreamReader may need to be inside a finally block around line 65 or so.

Reply:
Thanks Satheesh. Since this is just a test class that I used to resolve the opt-out issue, I will hold off on looking at that. I have noted it as a low priority that can be fixed as time permits at the end of a sprint.